pkg/tenant/api/v1/api.go

```diff
@@ -39,13 +39,15 @@ type TenantClient interface {
 
 	TenantProjectList(ctx context.Context, id string, in *PageQuery) (*TenantProjectPage, error)
 	TenantProjectCreate(ctx context.Context, id string, in *Project) (*Project, error)
+	TenantProjectPatch(ctx context.Context, tenantID, projectID string, in map[string]interface{}) (*Project, error)
```

Comment:
`in` can still be a `*Project` -- just make sure that there are `json:"omitempty"` tags in the `Project` struct so that the fields aren't present when they're sent to the server. None of the editable fields can be set to empty string or empty ID.